src/main/java/org/ow2/proactive/procci/model/occi/platform/bigdata/Swarm.java

```diff
@@ -0,0 +1,146 @@
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.AGENTS_IP;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.AGENTS_IP_NAME;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.HOST_IP;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.HOST_IP_NAME;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.MACHINE_NAME;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.MACHINE_NAME_NAME;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.MASTER_IP;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.MASTER_IP_NAME;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.NETWORK_NAME;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataAttributes.NETWORK_NAME_NAME;
+import static org.ow2.proactive.procci.model.occi.platform.bigdata.constants.BigDataIdentifiers.SWARM_MODEL;
+import static org.ow2.proactive.procci.model.occi.platform.constants.PlatformAttributes.STATUS_NAME;
+
+@Getter
+@ToString
+@EqualsAndHashCode(callSuper = true)
+public class Swarm extends Component {
+
+    private final String hostIp;
+    private final String masterIp;
+    private final List<String> agentsIp;
+    private Optional<String> machineName;
+    private Optional<String> networkName;
+
+    public Swarm(Optional<String> url,
+            Kind kind,
+            Optional<String> title,
+            List<Mixin> mixins,
+            Optional<String> summary,
+            List<Link> links,
+            Optional<Status> status,
+            Optional<String> machineName,
+            String hostIp,
+            String masterIp,
+            List<String> agentsIp,
+            Optional<String> networkName
+    ) {
+        super(url, kind, title, mixins, summary, links, status);
+        this.machineName = machineName;
+        this.hostIp = hostIp;
+        this.masterIp = masterIp;
+        this.agentsIp = agentsIp;
+        this.networkName = networkName;
+    }
+
+    public static Set<Attribute> createAttributeSet() {
+        Set<Attribute> attributeSet = new HashSet<>();
+        attributeSet.addAll(Component.createAttributeSet());
+        attributeSet.add(MACHINE_NAME);
+        attributeSet.add(HOST_IP);
+        attributeSet.add(MASTER_IP);
+        attributeSet.add(AGENTS_IP);
+        attributeSet.add(NETWORK_NAME);
+        return attributeSet;
+    }
+
+    /**
+     * Convert OCCI swarm to Proactive Cloud Automation Compute
+     *
+     * @param actionType is the action to apply on the swarm
+     * @return the proactive cloud automation model for the swarm
+     */
+    public Model toCloudAutomationModel(String actionType) {
```

Comment:
I think that each conversion is link to the class so it is relevant that each class implement the method. Indeed each class that can be translated into workflows has to implement the toCloudAutomationModel method. However I thought about adding an interface which would gather those translation methods.  What do you think ?